src/Main.scala

```diff
@@ -0,0 +1,35 @@
+import java.util.concurrent.{ExecutorService, Executors, PriorityBlockingQueue}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+object Main {
+
+  object TimeOrder extends Ordering[ScheduledTask] {
+    override def compare(x: ScheduledTask, y: ScheduledTask): Int = x.timestamp compare y.timestamp
+  }
+
+  val NUM_OF_THREAD = 5
+  val SECOND_IN_MILLI = 1000
+
+  def main(args: Array[String]): Unit = {
+
+    val ex = Executors.newFixedThreadPool(NUM_OF_THREAD)
+    implicit val ec = ExecutionContext.fromExecutorService(ex)
+
+    val size: Int = Try(args(0)).map(_.toInt).getOrElse(100)
+    val minHeap = new PriorityBlockingQueue[ScheduledTask](size, TimeOrder)
+    val scheduler = new SchedulerImp(minHeap, ec)
```

Comment:
don't need to pass `ec`. Let the compiler find the `implicit` execution context in this scope.